client/rpc/src/state/mod.rs

```diff
@@ -318,19 +319,26 @@ where
 			.map_err(Into::into)
 	}
 
-	fn subscribe_runtime_version(&self, sink: PendingSubscription) {
-		self.backend.subscribe_runtime_version(sink)
+	fn subscribe_runtime_version(&self, sink: SubscriptionSink) -> SubscriptionResult {
+		self.backend.subscribe_runtime_version(sink);
+		Ok(())
 	}
 
-	fn subscribe_storage(&self, sink: PendingSubscription, keys: Option<Vec<StorageKey>>) {
+	fn subscribe_storage(
+		&self,
+		mut sink: SubscriptionSink,
+		keys: Option<Vec<StorageKey>>,
+	) -> SubscriptionResult {
 		if keys.is_none() {
 			if let Err(err) = self.deny_unsafe.check_if_safe() {
 				let _ = sink.reject(JsonRpseeError::from(err));
-				return
+				// NOTE: `SubscriptionEmptyError` is introduced for ergonomic purposes.
```

Comment:
is this note just pointing out why we return `Result`s now? Probably not needed (we don't bring it up anywhere else either) :)